components/lsp/LSP.tsx

```diff
@@ -80,6 +82,15 @@ const LSP: FC<Props> = ({
     }
   }, [data.expirationTimestamp, currentTime, contractState]);
 
+  // Coverage in case there is a transition between states and the price settles.
+  useEffect(() => {
+    if (
+      data.contractState > ContractState.ExpiredPriceRequested &&
+      settleButtonDisabled
+    )
+      setSettleButtonDisabled(true);
```

Comment:
Can you double check this logic, its only being set true if its already true

Reply:
Yup, got it.